cookbooks/postgresql/recipes/server_install.rb

```diff
@@ -41,43 +39,50 @@
   notifies :run, "execute[update-apt]", :immediately
 end
 
-bash "delete-postgresql-service-from-package" do
-  code %Q{
-    [[ -n $(systemctl status postgresql | grep "Loaded.*/lib/systemd/system/postgresql.service") ]] && systemctl stop postgresql && rm /lib/systemd/system/postgresql.service
-  }
-  returns [0, 1]
-  action :nothing
-  only_if { !File.exist?("#{node['postgresql']['datadir']}/postmaster.pid") }
-  notifies :run, "execute[reload-systemd]", :immediately
+directory "/tmp/src/postgresql" do
+  action :create
+  recursive true
 end
 
-# this ruby block handles if the lock version file is set
+# install postgresql dependencies
+package "postgresql-common"
+package "postgresql-client-common"
+package "libpq-dev"
+package "libpq5"
+
+# This ruby block handles if the lock version file is set
 # It needs to be done like this since the file isn't present during the compile
 # phase on first runs on new instances booted from snapshots
-=begin # remove lock version feature for now
+# If a lock version file exists, use the version to set the variables
+# on template "/tmp/src/postgresql/install.sh"
 ruby_block 'check lock version' do
   block do
     if File.exists?(node['lock_version_file'])
-      install_version  = %x{cat #{node['lock_version_file']}}.strip
-      package_version = known_versions.detect {|v| v =~ /^#{install_version}/}
-      if package_version.nil?
-        Chef::Log.info "Chef does not know about PostgreSQL version #{install_version}"
-        exit(1)
-      end
-
-      run_context.resource_collection.find(:package => "postgresql-#{postgres_version}").version "#{package_version}-*"
-      run_context.resource_collection.find(:package => "postgresql-#{postgres_version}").not_if "apt-cache policy postgresql-#{postgres_version} | grep -E 'Installed.*#{package_version}-'"
+      install_version = %x{cat #{node['lock_version_file']}}.strip
+    else
+      install_version = node['postgresql']['latest_version']
     end
+    package_version = known_versions.detect {|v| v =~ /^#{install_version}/}
+    if package_version.nil?
+      Chef::Log.info "Chef does not know about PostgreSQL version #{install_version}"
+      exit(1)
+    end
+
+    run_context.resource_collection.find(:template => "/tmp/src/postgresql/install.sh").variables package_version: package_version, postgres_version: postgres_version
   end
 end
-=end
 
-package "postgresql-#{postgres_version}" do
-  notifies :run, "bash[delete-postgresql-service-from-package]", :immediately
+template "/tmp/src/postgresql/install.sh" do
```

Comment:
Will this be run twice? It's called from `ruby_block[check lock version]` and then from here?

Reply:
It will only run once, on the execute block (not here or the ruby_block). The ruby_block just sets the variables on the template.

Reply:
Ah, got it.